jib-core/src/main/java/com/google/cloud/tools/jib/docker/DockerClient.java

```diff
@@ -168,16 +174,41 @@ public String load(ImageTarball imageTarball) throws InterruptedException, IOExc
    * @throws InterruptedException if the 'docker save' process is interrupted
    * @throws IOException if creating the tarball fails
    */
-  public void save(ImageReference imageReference, Path outputPath)
+  public void save(
+      ImageReference imageReference,
+      Path outputPath,
+      ProgressEventDispatcher.Factory progressEventDispatcherFactory)
```

Comment:
I guess I prefer de-coupling anything related to `ProgressEventDispatcher` from `DockerClient.` I've checked that, currently, `ProgressEventDispatcher` lives in the `steps` packages and doesn't go over the boundary of it. For example, `RegistryClient.pushBlob` instead accepts `writtenByteCountListener` as type `Consumer<Long>`. I think de-coupling has a merit, since otherwise, to call `DockerClient.save()`, the caller must supply some `ProgressEventDispatcher.Factory`. I think `DockerClient` can be free of its involvement and just do the core work. `SaveDockerStep` can prepare the throttled written byte consumer.